### PR TITLE
Inverse of zero and infinite quaternion

### PR DESCRIPTION
**Pull request by arighnac**
In lieu of issue #13690 and PR #13693  , this modified code throws a Value Error whenever we try to find out the inverse of zero and infinite quaternion.

The tests for zero quaternion have been added.

Fixes the unsolved issue of PR #13693

### CODE REVIEW

**Comment by ArighnaIITG** [comment]
If approved , I can add the tests for infinite quaternion.  @asmeurer  @gxyd

File: sympy/algebras/quaternion.py [hunk 1]
```diff
@@ -304,6 +305,10 @@ def inverse(self):
         """Returns the inverse of the quaternion."""
         q = self
         return conjugate(q) * (1/q.norm()**2)
+        if not q.norm() :
```
Comment (gxyd) [inline sympy/algebras/quaternion.py]: We would be needing a lower level fix than this.

**Review by gxyd — COMMENTED** [review]
See the definition of Quaternion, it only allows 'a, b, c, d' all to be only real number, while `oo` is not.

File: sympy/algebras/quaternion.py [hunk 2]
```diff
@@ -304,6 +305,10 @@ def inverse(self):
...
+            raise ValueError("Cannot compute inverse of a quaternion with zero norm")
+        if q.norm() == oo :
```
Comment (gxyd) [inline sympy/algebras/quaternion.py]: You should only focus on raising an error for infinite quaternion. Raising error for inverse of a zero quaternion is being fixed by PR #13693.

**Comment by ArighnaIITG** [comment]
Then how do I go on about the infinite case?

**Comment by gxyd** [comment]
I think we should raise ValueError even for creation of a quaternion (I am not talking about inverse) that is passed with non-real values of 'a, b, c, d' (any one of them). Since the definition of Quaternion restricts them to be real numbers.

That would include the case of `Quaternion(oo, oo, oo, oo)`.

**Comment by jksuom** [comment]
> Since the definition of Quaternion restricts them to be real numbers.

This definition is more general. A quaternion algebra can be defined over any commutative ring.

**Comment by gxyd** [comment]
Hmm, so seeing from the wikipedia page, I can understand that what I am referring to is basically Hamilton's quaternions (which imposes 'a, b, c, d' to be real numbers only). https://en.wikipedia.org/wiki/Quaternion_algebra is what sympy actually implements.

**Comment by ArighnaIITG** [comment]
So, then how do I do it? Do I go on with the ValueError thrown when defined as infinite, or something else?
And ,please help me with writing the test on this issue.

**Comment by jksuom** [comment]
>  ValueError thrown when defined as infinite

This should probably be done when a quaternion is created. Hence only the zero norm case would be checked in `inverse`. (Note that, in general, a non-zero quaternion may have zero norm, though not in the case of Hamilton's quaternions.) The test itself should probably be `q.norm().is_zero`.

**Comment by ArighnaIITG** [comment]
So, do I need to create any separate modification for Infinite?

**Comment by jksuom** [comment]
> do I need to create any separate modification for Infinite

I don't that that would be necessary. We should not expect having infinite quaternions.

**Pull request merged**